modules/library/render/src/main/java/org/geotools/renderer/lite/StreamingRenderer.java

```diff
@@ -1067,10 +1088,46 @@ Query getStyleQuery(
             List<ReferencedEnvelope> envelopes = null;
             // enable advanced projection handling with the updated map extent
             if (isAdvancedProjectionHandlingEnabled()) {
+                Map projectionHints = new HashMap();
+                if (isAdvancedProjectionDensificationEnabled()
+                        && !CRS.equalsIgnoreMetadata(featCrs, mapCRS)) {
+                    ReferencedEnvelope targetEnvelope = envelope;
+                    ReferencedEnvelope sourceEnvelope = targetEnvelope.transform(featCrs, true);
```

Comment:
I think you are talking of ProjectionHandler.transformEnvelope. I can:
 * externalize it, or
 * make it public (currently protected), so that I can use it from StramingRenderer